label_studio/tests/data_import/test_persistent_storage_data.py

```diff
@@ -0,0 +1,336 @@
+from unittest import mock
+from unittest.mock import Mock
+
+import pytest
+from data_import.api import DownloadStorageData
+from data_import.models import FileUpload
+from django.conf import settings
+from django.http import HttpResponse
+from organizations.models import Organization
+from rest_framework import status
+from rest_framework.test import APIRequestFactory
+from users.models import User
+
+"""
+Comprehensive test suite for DownloadStorageData API
+
+This test module validates the complete functionality of the DownloadStorageData view,
+which handles secure file downloads from persistent storage (S3, GCS, Azure, etc.).
+
+## Test Coverage:
+
+### Authentication & Authorization Tests:
+- Missing filepath parameter validation
+- Invalid filepath security checks  
+- FileUpload permission validation
+- Organization-based avatar access control
+- Cross-organization access prevention
+
+### File Serving Mode Tests:
+- **NGINX Mode**: X-Accel-Redirect header generation for high-performance serving
+- **Direct Mode**: RangedFileResponse streaming with range request support
+- Mode switching via USE_NGINX_FOR_UPLOADS setting
+
+### File Type Support Tests:
+- Upload files (settings.UPLOAD_DIR) with permission checking
+- Avatar files (settings.AVATAR_PATH) with organization validation
+- Content-Type detection for various formats (PDF, MP3, MP4, JPG, unknown)
+- Content-Disposition headers (inline vs attachment)
+
+### Edge Case & Security Tests:
+- URL encoding/decoding support
+- NGINX redirect URL construction
+- Mock response handling for different scenarios
+- Error response codes (403, 404)
+
+## Test Structure:
+Each test method focuses on a specific aspect of the API:
+1. Sets up mock objects and request data
+2. Calls the DownloadStorageData.get() method
+3. Validates response status, headers, and behavior
+4. Verifies security constraints and permissions
+
+## Mocking Strategy:
+- FileUpload.objects.filter() for database queries
+- User.objects.filter() for avatar access
+- RangedFileResponse for direct file serving
+- Settings overrides for mode switching
+- Organization membership checks
+
+This comprehensive test suite ensures the API correctly handles all file types,
+security scenarios, and serving modes while maintaining backward compatibility.
+"""
+
+pytestmark = pytest.mark.django_db
+
+
+class TestDownloadStorageData:
+    @pytest.fixture
+    def api_factory(self):
+        return APIRequestFactory()
+
+    @pytest.fixture
+    def user(self):
+        """Create a test user with organization"""
+        org = Organization.objects.create(title='Test Org')
```

Comment:
I think we need to use either API or factories for data preparation in the tests, not the direct orm requests.